src/root/POSRootChainManager.ts

```diff
@@ -101,6 +101,17 @@ export default class POSRootChainManager extends ContractsBase {
     return this.posRootChainManager.methods.processedExits(exitHash).call()
   }
 
+  async processReceivedMessage(contractAddress: address, txHash: string) {
+    const payload = await this.exitManager.buildPayloadForExitHermoine(txHash, MESSAGE_SENT_EVENT_SIG)
+    let rootTunnelContract = new this.web3Client.parentWeb3.eth.Contract(this.rootTunnelContractAbi, contractAddress)
+    return rootTunnelContract.methods.receiveMessage(payload)
+  }
+
+  async customPayload(txHash: string) {
```

Comment:
Can we also add eventSig as an argument here? 
```suggestion
  async customPayload(txHash: string, eventSig) {
```